pydantic/generics_utils.py

```diff
@@ -0,0 +1,58 @@
+import sys
+import typing
+from types import prepare_class, resolve_bases
+from typing import Any
+
+TypeVarType = Any  # since mypy doesn't allow the use of TypeVar as a type
+Parametrization = typing.Mapping[TypeVarType, type[Any]]
+
+if typing.TYPE_CHECKING:
+    from pydantic import BaseModel
+
+    class GenericBaseModel(BaseModel):
+        __parameters__: typing.ClassVar[typing.Tuple[TypeVarType, ...]]
```

Comment:
This type-checking-only `GenericBaseModel` class, along with the `is_generic_model` typeguard below, are just there to help address the fact that mypy doesn't like when I access `__parameters__` on a class that inherits from both `BaseModel` and `typing.Generic`. This approach seemed to handle it correctly with minimal mypy ignores.